tests/test_compare_dictconfig_vs_dict.py

```diff
@@ -0,0 +1,544 @@
+        """Create a DictConfig instance from the given data"""
+        cfg: DictConfig = DictConfig(content=python_dict)
+        if struct_mode:
+            OmegaConf.set_struct(cfg, True)
+        return cfg
+
+    def test__setitem__(
+        self, python_dict: Any, cfg: DictConfig, key: Any, struct_mode: bool
+    ) -> None:
+        """Ensure that __setitem__ has same effect on python dict and on DictConfig."""
+        if struct_mode and key not in cfg:
+            with pytest.raises(ConfigKeyError):
+                cfg[key] = "sentinel"
+        else:
+            python_dict[key] = "sentinel"
+            cfg[key] = "sentinel"
+            assert python_dict == cfg
+
+    def test__getitem__(self, python_dict: Any, cfg: DictConfig, key: Any) -> None:
+        """Ensure that __getitem__ has same result with python dict as with DictConfig."""
+        try:
+            result = python_dict[key]
+        except KeyError:
+            with pytest.raises(ConfigKeyError):
+                cfg[key]
+        else:
+            assert result == cfg[key]
+
+    def test__delitem__(
+        self, python_dict: Any, cfg: DictConfig, key: Any, struct_mode: bool
+    ) -> None:
+        """Ensure that __delitem__ has same result with python dict as with DictConfig."""
+        if struct_mode:
+            with pytest.raises(ConfigTypeError):
+                del cfg[key]
+        else:
+            try:
+                del python_dict[key]
+                assert key not in python_dict
+            except KeyError:
+                with pytest.raises(ConfigKeyError):
+                    del cfg[key]
+            else:
+                del cfg[key]
+                assert key not in cfg
+
+    def test__contains__(self, python_dict: Any, cfg: Any, key: Any) -> None:
+        """Ensure that __contains__ has same result with python dict as with DictConfig."""
+        assert (key in python_dict) == (key in cfg)
+
+    def test__eq__(self, python_dict: Any, cfg: Any, key: Any) -> None:
+        assert python_dict == cfg
+
+    @pytest.mark.parametrize("use_default", [True, False])
+    def test_get(
+        self, python_dict: Any, cfg: DictConfig, key: Any, use_default: bool
+    ) -> None:
+        """Ensure that __getitem__ has same result with python dict as with DictConfig."""
+        if use_default:
+            assert python_dict.get(key, "DEFAULT") == cfg.get(key, "DEFAULT")
+        else:
+            assert python_dict.get(key) == cfg.get(key)
+
+    @pytest.mark.parametrize("use_default", [True, False])
+    def test_pop(
+        self,
+        python_dict: Any,
+        cfg: DictConfig,
+        key: Any,
+        use_default: bool,
+        struct_mode: bool,
+    ) -> None:
+        """Ensure that __getitem__ has same result with python dict as with DictConfig."""
+        if struct_mode:
+            with pytest.raises(ConfigTypeError):
+                cfg.pop(key)
```

Comment:
If there are behavior changes, it's best to use a different test method, e.g. `test_pop_dict_config_struct_mode`.